features/opennms-es-rest/main-module/src/main/java/org/opennms/plugins/elasticsearch/rest/archive/OpenNMSHistoricEventsToEs.java

```diff
@@ -0,0 +1,160 @@
+package org.opennms.plugins.elasticsearch.rest.archive;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+
+import org.opennms.netmgt.events.api.EventForwarder;
+import org.opennms.plugins.elasticsearch.rest.archive.OnmsRestEventsClient;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.Parm;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** 
+ * takes events from OpenNMS historic events through the OpennMS
+ * rest interface and sends them to elastic search using the same queue
+ * as live incoming events use.
+ * @author admin
+ *
+ */
+public class OpenNMSHistoricEventsToEs {
+	private static final Logger LOG = LoggerFactory.getLogger(OpenNMSHistoricEventsToEs.class);
+
+	private static final int EVENT_RETREIVAL_LIMIT=10; // retrieve and process max 10 events at a time
+
+	private String onmsUrl="http://localhost:8980";
+
+	private String onmsUserName="admin";
+
+	private String onmsPassWord="admin";
+
+	private Integer limit=10;
+
+	private Integer offset=0;
+
+	private EventForwarder eventForwarder=null;
+	
+	private boolean useNodeLabel=true;
+
+	public String getOnmsUrl() {
+		return onmsUrl;
+	}
+
+	public void setOnmsUrl(String onmsUrl) {
+		this.onmsUrl = onmsUrl;
+	}
+
+	public String getOnmsUserName() {
+		return onmsUserName;
+	}
+
+	public void setOnmsUserName(String onmsUserName) {
+		this.onmsUserName = onmsUserName;
+	}
+
+	public String getOnmsPassWord() {
+		return onmsPassWord;
+	}
+
+	public void setOnmsPassWord(String onmsPassWord) {
+		this.onmsPassWord = onmsPassWord;
+	}
+
+	public EventForwarder getEventForwarder() {
+		return eventForwarder;
+	}
+
+	public void setEventForwarder(EventForwarder eventForwarder) {
+		this.eventForwarder = eventForwarder;
+	}
+
+	public Integer getLimit() {
+		return limit;
+	}
+
+	public void setLimit(Integer limit) {
+		this.limit = limit;
+	}
+
+	public Integer getOffset() {
+		return offset;
+	}
+
+	public void setOffset(Integer offset) {
+		this.offset = offset;
+	}
+	
+	public boolean getUseNodeLabel() {
+		return useNodeLabel;
+	}
+
+	public void setUseNodeLabel(boolean useCache) {
+		this.useNodeLabel = useCache;
+	}
+	
+
+	public OpenNMSHistoricEventsToEs(){
+		super();
+	}
+
+	/**
+	 * sends events to elastic search returns true if successful
+	 * @return
+	 */
+	public String sendEventsToEs(){
+
+		OnmsRestEventsClient onmsRestEventsClient = new OnmsRestEventsClient();
```

Comment:
The intent of this is to allow us to import old events into Elastic search from any OpenNMS instance using ReST. So we can import locally and remotely. This allows us to test at the customer site without changing their current Meridian system.
Secondly I looked at the eventdao and the only usable method getEventsAfterDate(final List<String> ueiList, final Date date)  seems to require an explicit list of uei's (i.e. no wild cards for uei). So difficult to get assess to a paged list of all events in the system. Using the ReST interface was the simplest way to get to a paged list of all of the events  and it allows import from external systems.